lib/IRGen/IRGenSIL.cpp

```diff
@@ -4731,11 +4731,18 @@ void IRGenSILFunction::visitDebugValueInst(DebugValueInst *i) {
   VarInfo->Name = getVarName(i, IsAnonymous);
   DebugTypeInfo DbgTy;
   SILType SILTy;
-  if (auto MaybeSILTy = VarInfo->Type)
+  bool IsFragmentType = false;
+  if (auto MaybeSILTy = VarInfo->Type) {
     // If there is auxiliary type info, use it
     SILTy = *MaybeSILTy;
-  else
+  } else {
     SILTy = SILVal->getType();
+    if (auto Expr = VarInfo->DIExpr)
+      if (auto n = Expr.getNumElements())
+        if (Expr.getElement(n - 1).getAsOperator() ==
+            SILDIExprOperator::Fragment)
```

Comment:
I think `SILDebugInfoExpression::hasFragment` can be used here

Reply:
Much better!